CMakeLists.txt

```diff
@@ -86,6 +86,9 @@ if (ONEDPL_USE_PARALLEL_POLICIES)
                     execute_process(COMMAND ${FIND_GXX_EXE} -dumpfullversion OUTPUT_VARIABLE _onedpl_gxx_version)
                     string(REPLACE "\." "0" _onedpl_gxx_version ${_onedpl_gxx_version})
                     target_compile_definitions(oneDPL INTERFACE TBB_USE_GLIBCXX_VERSION=${_onedpl_gxx_version})
+                    if (_onedpl_gxx_version GREATER_EQUAL 80100 AND CMAKE_CXX_STANDARD GREATER_EQUAL 17)
```

Comment:
1. You're doing things under `if (CMAKE_CXX_COMPILER_ID STREQUAL Clang)`, but the handling should be universal.
2. I'd suggest the following logic (variable names are just examples):
```cmake
if (CMAKE_CXX_STANDARD GREATER_EQUAL 17 AND
    (GCC_VERSION_DETECTED AND GCC_VERSION GREATER_EQUAL 8.1) OR
    (NOT GCC_VERSION_DETECTED)
   )
```

Reply:
Code is changed, could you please check it?